### DISCUSSION TITLE
Should 'reply_markup' field exist in simple text message ?

### DISCUSSION THREAD

**Discussion by kolyande** [reo7sp/tgbot-cpp#248] [Q&A] [answered]
Hello, and Thanks to all contributors!
I found that, the default value for argument 'replyMarkup' in method Api::sendMessage https://github.com/reo7sp/tgbot-cpp/blob/d2bf07fee0ba828a837ca181c396eba0e8313b0f/include/tgbot/Api.h#L183 , the way it is checked inside method's body https://github.com/reo7sp/tgbot-cpp/blob/d2bf07fee0ba828a837ca181c396eba0e8313b0f/src/Api.cpp#L150 and the way it is parsed then in method TgTypeParser::parseGenericReply - ALL leads to existing of 'reply_markup' field with empty value "" for simple text messages like this: `{"chat_id":"1","text":"Hello","reply_markup":""}` .
Is it right behaviour ? Does it follow the Telegram bot API doc https://core.telegram.org/bots/api#message where this field is described as _Optional_ ?

**Comment by we-lu** [comment, marked as answer]
Thank you for the message. No, that is not the correct behaviour. The empty field is just unnecessary. I'm surprised I didn't see that before either. I created a pull request that should fix it.
